packages/paste-core/components/button/src/types.ts

```diff
@@ -1,42 +1,19 @@
-type ButtonTypes = 'submit' | 'button' | 'reset';
+// type ButtonTypes = 'submit' | 'button' | 'reset';
 export type ButtonSizes = 'small' | 'default' | 'icon' | 'reset';
 export type ButtonVariants = 'primary' | 'secondary' | 'destructive' | 'destructive_link' | 'link' | 'reset';
 export type ButtonStates = 'disabled' | 'loading' | 'default';
 export type ButtonTabIndexes = 0 | -1;
 
-export interface ButtonProps {
-  id?: never;
-  className?: never;
-  type?: ButtonTypes;
-  href?: string;
-  autoFocus?: boolean;
+export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
   as?: string;
-  tabIndex?: ButtonTabIndexes;
-
-  variant: ButtonVariants;
-  size?: ButtonSizes;
-  fullWidth?: boolean;
-
   children: React.ReactNode;
-
-  disabled?: boolean;
+  fullWidth?: boolean;
+  href?: string;
+  id?: never;
```

Comment:
I think we should probably allow id's honestly.

Reply:
Pulled `id` out since we get it with the html attributes.